### ISSUE TITLE
Make svg.path math operations faster using numba ?

### ISSUE THREAD

**Issue by mpage38** [regebro/svg.path#102]
I want to process a lot of SVG glyphs, in particular computing their bounding box. Presently, this operation is costly using svg.path. Have you considered using numba for making math operations faster ?

**Comment by regebro** [comment]
No, but it could be worth a try, the bezier bounding box calculations are pretty complex.

**Comment by regebro** [comment]
Quick testing shows a large amount of warnings that make no sense to me, and significantly slower runtimes. It's possible a numba expert can do better, but I will not.

**Comment by mpage38** [comment]
Also made some testing on my side. After profiling svg.path, I focused on optimizing the most time-consuming functions using numba, and it didn't show much improvements (10-15% in my tests). I then did the same operation using cython. This is much more promising. I got 10x improvement on the most time-consuming functions.
... But cython is another beast than python, with its own quirks.

**Comment by regebro** [comment]
Yeah, but that sounds like a much better way forward, at least. I haven't used cython since it was called pyrex, though. :-D

**Comment by regebro** [comment]
I decided to test Cython before closing this, and yes, seems like it could be effective, and it's too hard to implement. 

Some methods definitely need some rewriting, and switching to arrays instead of lists, etc. But even just with no code changes tests are measurably faster, maybe shaving off some 25% of the time or so.

I made a branch with the setup https://github.com/regebro/svg.path/tree/regebro-cython

**Comment by regebro** [comment]
OK, I looked more at it, and with things that really has the potential to speed up I didn't get any speed improvements, in fact it seems I'm not even running the pythonized version in those tests, and I don't know why.

So I'm giving up on this, I don't have the time to learn Cython properly.
